docs/implementations/moby.md

```diff
@@ -0,0 +1,79 @@
+# Moby
+
+The Moby client is the open source upstream of Docker, and the workflow has been widely copied.
+
+## API Workflow
+
+- Ping the v2 endpoint (`GET /v2/`), if 401 (unauthorized status), save values from the `WWW-Authenticate` header
+- Manually set scope to `repository:${repo}:${action}` with action = `pull`, `push`, or `pull,push`
+- If `from` query parameter seen on request, add additional scope with a `pull` action
+- Auth type (basic or token) is determined from v2 ping `WWW-Authenticate` header to set `Authorization` header on all requests to the repository
+- For basic auth, the `Authorization` header is set to `Basic $(base64 ${user}:${pass})`
+- For token auth:
+  - Extract the `realm` and `service` values from the v2 ping
+  - If scope has a new entry always get a token, otherwise use token from cache if still valid
+  - Use oauth if a refresh token is available to get token
+    - Set form values:
+      - `service`: from v2 ping
+      - `scope`: concatenated list (comma separated) of scopes
+      - `grant_type`: `refresh_token` if available, else `password`
+      - `refresh_token`: set if available
+      - `username`, `password`, `access_type=offline`: for password auth
+    - Request is set to the `realm` url
+    - Response is parsed for `access_token`, `refresh_token`, `expires_in`, `issued_at`, and `scope`
```

Comment:
is only JWT supported?

Reply:
I don't believe JWT is explicitly supported. It's not parsing the content of the access token itself, so it may be JWT or any other opaque value.